engine/src/main/java/org/terasology/persistence/serializers/GsonSerializer.java

```diff
@@ -52,47 +71,112 @@ public GsonSerializer() {
+     * @param object the object to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param writer The writer in which the JSON will be written to
+     */
     public <T> void writeJson(T object, TypeHandler<T> typeHandler, Writer writer) {
-        GsonPersistedData persistedData =
-                (GsonPersistedData) typeHandler.serialize(object, new GsonPersistedDataSerializer());
+        GsonPersistedData persistedData = (GsonPersistedData) typeHandler.serialize(object,
+                new GsonPersistedDataSerializer());
 
         gson.toJson(persistedData.getElement(), writer);
     }
 
+    /**
+     * Writes an object's serialized persisted data to an OutputStream
+     * 
+     * @see #writeJson(Object, TypeHandler, Writer)
+     * @param object the object to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param stream stream that the data will be written to
+     * @throws IOException will be thrown if there is an error writing to the stream
+     */
     public <T> void writeJson(T object, TypeHandler<T> typeHandler, OutputStream stream) throws IOException {
         try (Writer writer = new BufferedWriter(new OutputStreamWriter(stream))) {
             writeJson(object, typeHandler, writer);
         }
     }
 
+    /**
+     * Writes the object's persisted data to a File object
+     * 
+     * @see #writeJson(Object, TypeHandler, String)
+     * @param object the file to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param file file that the bytes will be written to
+     * @throws IOException gets thrown if there is an issue writing to the file
+     */
     public <T> void writeJson(T object, TypeHandler<T> typeHandler, File file) throws IOException {
         try (Writer writer = new BufferedWriter(new FileWriter(file))) {
             writeJson(object, typeHandler, writer);
         }
     }
 
-    public <T> void writeJson(T object, TypeHandler<T> typeHandler, String path) throws IOException {
-        writeJson(object, typeHandler, new File(path));
+    /**
+     * Writes the a the object's persisted data to a File located at a specified file name
+     * 
+     * @see #writeJson(Object, TypeHandler, String)
+     * @param object the object to be serialized
+     * @param typeHandler contains how the object will be serialized
+     * @param fileName the file name where the JSON will be written
+     * @throws IOException gets thrown if there is an error writing to the file at the specified location
+     */
+    public <T> void writeJson(T object, TypeHandler<T> typeHandler, String fileName) throws IOException {
+        writeJson(object, typeHandler, new File(fileName));
     }
 
+    /**
+     * Gets the PersistedData from a Reader object
+     * 
+     * @see #persistedDataFromJson(InputStream)
+     * @param reader Reader object that will be deserialized
```

Comment:
Like below, the _data_ read from the `Reader` will be deserialized, not the `Reader` itself.

Reply:
Once again, I thought it could be implied that it meant the contents. I will change it